docs/data/data-grid/column-pinning/column-pinning.md

```diff
@@ -51,25 +53,65 @@ Use it together with `onPinnedColumnsChange` to know when a column is pinned or
 
 {{"demo": "ControlPinnedColumns.js", "bg": "inline"}}
 
-## Blocking column unpinning
+## Disable column pinning
+
+### For all columns
+
+Column pinning is enabled by default, but you can easily disable this feature by setting the `disableColumnPinning` prop.
+
+```tsx
+<DataGridPro disableColumnPinning />
+```
+
+### For some columns
 
-It may be desirable to not allow a column to be unpinned.
-The only thing required to achieve that is to hide the buttons added to the column menu.
-This can be done in two ways:
+To disable the pinning of a single column, set the `pinnable` property in `GridColDef` to `false`.
 
-1. Per column, by setting `pinnable` to `false` in each `GridColDef`:
+```tsx
+<DataGridPro columns={[{ field: 'id', pinnable: false }]} /> // Default is `true`.
+```
 
-   ```tsx
-   <DataGrid columns={[{ field: 'id', pinnable: false }]} /> // Default is `true`.
-   ```
+### Pin non-pinnable columns programmatically
 
-2. By providing a custom menu, as demonstrated below:
+It may be desirable to allow one or more columns to be pinned (or unpinned) programmatically which don't support pinning (i.e. columns for which prop `disableColumnPinning = true` or `colDef.pinnable = false`).
```

Comment:
"which don't support pinning": that's a bit confusing, I'd rather phrase it: "which can't be unpinned by the user".